### PR TITLE
Make all UriBeacon notifications public on Lollipop

### PR DESCRIPTION
**Pull request by cco3**
There doesn't seem to be any reason why public beacons would give
anything other than public notifications.  This change makes beacons
publicly visible on the Android 5.0 lock screen when the user has
"hide sensitive notification content" selected.

However, this change keeps urls discovered through mdns or ssdp as
private.

This a continuation of #299 because I can't seem to reopen it.

### CODE REVIEW

**Comment by g-ortuno** [comment]
@cco3 Could you test this in a N7 and a N4? We've seen inconsistent behavior in the past and I just want to make sure we aren't breaking other devices again.

**Comment by cco3** [comment]
Which N7?  There are 4 of them, haha (grouper, tilapia, flo, deb).  Also, where did the inconsistent behavior lie?

**Comment by g-ortuno** [comment]
Inconsistencies are, sadly, all over the place. In this case I would make sure the notifications are showing up in N7, since that's where we've have problems in the past. Testing on < 21 would also be appreciated.

**Comment by g-ortuno** [comment]
Why do you exactly need an extra parameter? You can just add the url to the `publicUrls` hash set before you call `findUrlMetadata()`.

**Comment by cco3** [comment]
Because that url isn't necessarily the same one that's set later (the one returned by the metadata server).

**Comment by g-ortuno** [comment]
What makes you say that?

**Comment by cco3** [comment]
We send a url to the server in a json object, it gives us a json object back with a url in it.  We don't hang on to the url, so isn't it possible that the service could give us a new URL?  For example, couldn't it give us a lengthened version of a short url?

**Comment by g-ortuno** [comment]
If you step through the code you can see that `onUrlMetadataReceived` is called with the original url as a parameter and the lengthened url is in `urlMetadata.siteUrl`

**Comment by cco3** [comment]
Ok, so the "id" field in the metadata response is guaranteed to be the original url?

**Comment by g-ortuno** [comment]
Yes.
https://github.com/google/physical-web/blob/master/web-service/main.py#L104

I've seen it being use like that throughout the code.

**Comment by cco3** [comment]
Done.

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UriBeaconDiscoveryService.java [hunk 1]
```diff
@@ -417,6 +422,11 @@ private void updateNearbyBeaconNotification(boolean single, String url, int noti
         .setContentText(description)
         .setPriority(NOTIFICATION_PRIORITY)
         .setContentIntent(pendingIntent);
+    if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+      if (mPublicUrls.contains(url)) {
+        builder.setVisibility(NOTIFICATION_VISIBILITY);
```
Comment (g-ortuno) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UriBeaconDiscoveryService.java]: Should be [VISIBILITY_PUBLIC](http://developer.android.com/reference/android/app/Notification.html#VISIBILITY_PUBLIC).

Comment (g-ortuno) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UriBeaconDiscoveryService.java]: Please test next time.

**Comment by g-ortuno** [comment]
One comment.

File: android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UriBeaconDiscoveryService.java (hunk 1 above)
Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UriBeaconDiscoveryService.java]: Not sure what you mean.  This is VISIBILITY_PUBLIC (See L104)

Comment (g-ortuno) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UriBeaconDiscoveryService.java]: You're completely right. Sorry about that. Still I tested it on my Nexus 5 and I don't see any changes in behavior...

Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UriBeaconDiscoveryService.java]: OK, I'll try it again.

Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UriBeaconDiscoveryService.java]: Works right for me.  Is it showing up as content hidden for you?

Comment (g-ortuno) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UriBeaconDiscoveryService.java]: Yes. I tried it in two ways:

1. Set `When device is locked` to `Hide sensitive notification content`
2. Turned on `Sensitive` from Apps Notifications Physical Web

In both cases I get `Contents hidden`.

Comment (g-ortuno) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UriBeaconDiscoveryService.java]: How many beacons do you have around you?

Comment (cco3) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UriBeaconDiscoveryService.java]: I just tried with 1, 2, and many beacons.  Everything looks correct.

I think if you set "sensitive" on the app, then it considers all the notifications sensitive regardless of what the app reports.  Make sure that is turned off and only the lock screen setting itself is set to hide sensitive notification content.

Comment (g-ortuno) [inline android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UriBeaconDiscoveryService.java]: Weird. Updating the device fixed the issue...

**Comment by cco3** [comment]
Everything works correctly on Nexus4/Nexus7(2013)...note that neither of these have 5.1 yet, so there is no hide sensitive notification content option.

**Pull request merged**